crates/bevy_ecs/src/world/mod.rs

```diff
@@ -1136,12 +1136,24 @@ impl World {
         }
     }
 
+    pub fn clear_all(&mut self) {
+        self.clear_entities();
+        self.clear_resources();
+    }
+
     pub fn clear_entities(&mut self) {
         self.storages.tables.clear();
         self.storages.sparse_sets.clear();
         self.archetypes.clear_entities();
         self.entities.clear();
     }
+
+    pub fn clear_resources(&mut self) {
+        let resource_archetype = self.archetypes.resource_mut();
+        for column in resource_archetype.unique_components.values_mut() {
+            column.clear();
+        }
```

Comment:
I imagine this clears internal Bevy resources as well? Should we document that? I think it could be an easy footgun for those who don't understand it.

Reply:
Yeah I'm sold we should document all three of these methods before merging.

They're pretty powerful footguns, and mostly useful for multi-world stuff or testing.